### ISSUE TITLE
Allow separate configuration of http_opts used for AWS metadata

### ISSUE THREAD

**Issue by jszwedko** [ex-aws/ex_aws#598]
Currently it appears that there is no way to configure the HTTP client used for fetching AWS metadata ([specifically for fetching IAM credentials from the EC2 instance role](
https://github.com/ex-aws/ex_aws/blob/b123d369fa5d4b17d12d6dac2d411695a00363f4/lib/ex_aws/instance_meta.ex)) separate from the default configuration used by `ex_aws`.

This is possible for individual AWS services ([as mentioned](https://github.com/ex-aws/ex_aws/blob/b123d369fa5d4b17d12d6dac2d411695a00363f4/lib/ex_aws/config.ex#L22-L37)) via something like:

```
config :ex_aws, :dynamodb,
  http_opts: [
    pool: :ex_aws_dynamodb
]
```

I believe it would be nice to be able to set configuration separately for metadata requests. Specifically, I'd like to separate the metadata requests to their own hackney pool to avoid them competing for connections with other requests (as we've seen this can lead to starvation in extreme cases).

Opening an issue first to gauge how receptive maintainers would be to a change like this.

I'm thinking something as simple allowing configuration that looks like the following:

```
config :ex_aws, :metadata,
  http_opts: [
    pool: :ex_aws_metadata
]
```

**Comment by benwilson512** [comment]
Sounds great, PR welcome!

**Comment by TheoLeanse** [comment]
Hey! did this get resolved and if not, are you still looking for help? :)

**Comment by bernardd** [comment]
Nope, I don't think there's been any movement on it. And help is absolutely always welcome :)

**Comment by bernardd** [comment]
Resolved by #765 - thanks @TheoLeanse!
